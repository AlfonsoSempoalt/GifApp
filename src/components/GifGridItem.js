import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ url, title, id }) => {
  return (
    <section className="card">
      <img src={url} alt={title}></img>
      <h4>{title}</h4>
    </section>
  );
};

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
