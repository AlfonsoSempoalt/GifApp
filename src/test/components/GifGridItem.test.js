import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import { describe, expect } from '@jest/globals';

describe('tests to <GifGridItem /> with enzyme', () => {
  const url = 'testing.com';
  const title = 'test';

  let wrapper = shallow(<GifGridItem url={url} title={title} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem url={url} title={title} />);
  });

  test('should match the wrapper with the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('the title should a string with text', () => {
    const testingTitle = wrapper.find('h4').text();
    expect(testingTitle).toBe(title);
  });

  test('should have an url matching with the props url and must have an alt the img', () => {
    const testingIMG = wrapper.find('img');
    const urlImg = testingIMG.prop('src');
    const altImg = testingIMG.prop('alt');

    expect(urlImg).toEqual(url);
    expect(altImg).toEqual(title);
  });

  test('should have an animate_fadeIn class', () => {
    const testingSection = wrapper.find('section');
    const classNameProperty = testingSection.prop('className');
    

    expect(classNameProperty.includes('animate__fadeIn')).toEqual(true);
  });
  
});
