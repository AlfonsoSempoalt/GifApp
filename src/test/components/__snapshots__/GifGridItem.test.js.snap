// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <GifGridItem /> with enzyme should match the wrapper with the snapshot 1`] = `
<section
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test"
    src="testing.com"
  />
  <h4>
    test
  </h4>
</section>
`;
